### PR TITLE
Issue #2973 Che workspace like Python, PHP ... deployment fails

### LINKED ISSUE #2973: Che workspace like Python, PHP ... deployment fails
### General information

  * Minishift version: master
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Steps to reproduce

  1. make integration GODOG_OPTS="-tags addon-che"
  2. 
  3. 
  4. 

### Expected
Should go through

### Actual
Fails - https://ci.centos.org/job/minishift-nightly-centos/328/consoleFull
           https://ci.centos.org/job/minishift-nightly-centos/329/consoleFull

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by agajdosi**
Fixes #2973

### CODE REVIEW

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
So the converted the []byte to string and it gives:
```
[{"namespace":"che","status":"STOPPED","config":{"commands":[],"environments":{"default":{"recipe":{"type":"dockerimage","content":"registry.centos.org/che-stacks/spring-boot"},"machines":{"dev-machine":{"env":{},"servers":{"tomcat8-debug":{"attributes":{},"protocol":"http","port":"8000"},"springboot":{"attributes":{},"protocol":"http","port":"8080"}},"installers":["org.eclipse.che.exec","org.eclipse.che.terminal","org.eclipse.che.ws-agent","org.eclipse.che.ls.java"],"volumes":{"m2":{"path":"/home/user/.m2"},"javadata":{"path":"/home/user/jdtls/data"}},"attributes":{"memoryLimitBytes":"2147483648"}}}}},"projects":[],"defaultEnv":"default","name":"spring-boot-stack-test","attributes":{},"links":[]},"links":{},"temporary":false,"id":"workspace0qo7qkdx9uw3stiw","attributes":{"updated":"1541519660855","created":"1541519659784"}}]
``` 
the question is now why that cannot be converted into JSON, giving error:

```
error: json: cannot unmarshal array into Go value of type che.WorkspaceStatus
```

investigating more

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
retest this please

**Comment by anjannath** [comment]
@agajdosi Is the initial error of converting to JSON fixed now?

**Comment by agajdosi** [comment]
Fixed tests has passed: https://ci.centos.org/job/minishift-pr/5706/console. It still has problems with PHP sample code's return value, but that is bug for another issue (going to create it once this is merged).

Problem with Che was that sometimes cluster was not ready, Che tried to delete the containers and silently failed. To fix I have added 2 `oc delete` with retry statements to make sure all workspaces are deleted before running any new workspace.

**Comment by anjannath** [comment]
retest this please

**Comment by anjannath** [comment]
retest this

File: addons/che/templates/che-server-template.yaml [hunk 1]
```diff
@@ -96,8 +96,8 @@ objects:
             value: "openshift"
           - name: CHE_INFRA_KUBERNETES_BOOTSTRAPPER_BINARY__URL
             value: "${PROTOCOL}://che-${NAMESPACE}.${ROUTING_SUFFIX}/agent-binaries/linux_amd64/bootstrapper/bootstrapper"
-          - name: CHE_INFRA_KUBERNETES_MACHINE__START__TIMEOUT__MIN
-            value: "5"
+          - name: CHE_INFRA_KUBERNETES_WORKSPACE__START__TIMEOUT__MIN
+            value: "10"
```
Comment (anjannath) [inline addons/che/templates/che-server-template.yaml]: What does this time out do?

Comment (agajdosi) [reply addons/che/templates/che-server-template.yaml]: AFAIU it defines maximal time for workspace to be running. If for any reason the workspace is not ready in that time the start is stopped and the startup is reported as failed. Default value in Che is 7 minutes.

Comment (agajdosi) [reply addons/che/templates/che-server-template.yaml]: They changed the naming some time ago :rofl: https://github.com/eclipse/che/commit/806a6da184c069e8cc0f6953c7af97e82994132a

**Pull request merged**